src/SonarQube.Client/Api/Requests/V5_20/GetQualityProfileChangeLogRequest.cs

```diff
@@ -0,0 +1,62 @@
+﻿/*
+ * SonarQube Client
+ * Copyright (C) 2016-2018 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace SonarQube.Client.Api.Requests.V5_20
+{
+    public class GetQualityProfileChangeLogRequest : PagedRequestBase<DateTime>, IGetQualityProfileChangeLogRequest
+    {
+        [JsonProperty("profileKey")]
+        public virtual string QualityProfileKey { get; set; }
+
+        [JsonProperty("qualityProfile", DefaultValueHandling = DefaultValueHandling.Ignore), DefaultValue(null)]
+        public virtual string QualityProfileName
+        {
+            get { return null; }
+            set { /* to prevent the property from serialization we always return the default */ }
```

Comment:
Why do we want to this property and LanguageName always to be null and not to be serialized?

Reply:
Wt does not exist in this version of the API and there is no point sending it.

Reply:
I agree this adds some complexity, but unfortunately once a property is `JsonIgnore`d it cannot be un-ignored. Other options are:
- create completely new implementation (same code - duplication, not less complex)
- create base class without this property (more classes, not less complex)
- always send the property (not sure what will happen in future versions of SQ if we send invalid property and is generally not very clean imo)

Reply:
I think the current implementation is fine: the only problem was that it wasn't clear from the comment why it is like that.
I think adding your explanation above would be enough e.g. _Property does not exist in this version of the API so no point in sending it_